test/k8s-integration/cluster.go

```diff
@@ -269,29 +276,116 @@ func clusterUpGKE(gceZone, gceRegion string, numNodes int, imageType string) err
 		}
 	}
 
-	var cmd *exec.Cmd
-	cmdParams := []string{"container", "clusters", "create", *gkeTestClusterName,
-		locationArg, locationVal, "--num-nodes", strconv.Itoa(numNodes),
-		"--quiet", "--machine-type", "n1-standard-2", "--image-type", imageType}
-	if isVariableSet(gkeClusterVer) {
-		cmdParams = append(cmdParams, "--cluster-version", *gkeClusterVer)
+	if useStagingDriver {
+		accessToken, err := getAccessToken()
+		if err != nil {
+			return err
+		}
+
+		token := &oauth2.Token{AccessToken: string(accessToken)}
+		oauthClient := oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(token))
+		service, err := container.NewService(context.Background(), option.WithHTTPClient(oauthClient))
+		if err != nil {
+			return fmt.Errorf("failed to create a new service: %v", err)
+		}
+
+		// TODO: change to read CLOUDSDK_API_ENDPOINT_OVERRIDES_CONTAINER env var
+		service.BasePath = "https://test-container.sandbox.googleapis.com/" // e.g test, staging env URL
+
+		// TODO: honor the gkeNodeVersion flag
+		request := &container.CreateClusterRequest{
+			Cluster: &container.Cluster{
+				Name:             *gkeTestClusterName,
+				InitialNodeCount: int64(numNodes),
+				AddonsConfig: &container.AddonsConfig{
+					GcpFilestoreCsiDriverConfig: &container.GcpFilestoreCsiDriverConfig{Enabled: true},
+				},
+				ReleaseChannel: &container.ReleaseChannel{Channel: "RAPID"},
+				NodeConfig: &container.NodeConfig{
+					MachineType: "n1-standard-2",
+					ImageType:   imageType,
+				},
+			},
+		}
+
+		klog.Infof("Creating kubernetes e2e cluster on gke")
+		project, err := getCurrProject()
+		if err != nil {
+			return err
+		}
+		parent := fmt.Sprintf("projects/%s/locations/%s", project, locationVal)
+		klog.Infof("Creating cluster under parent path %s", parent)
+		op, err := service.Projects.Locations.Clusters.Create(parent, request).Do()
+		if err != nil {
+			return fmt.Errorf("failed to Create kubernetes e2e cluster on gke: %v", err)
+		}
+		err = waitForOp(service.Projects.Locations.Operations, parent, op)
+		if err != nil {
+			return fmt.Errorf("WaitFor Cluster Create operation failed: %v", err)
+		}
+
+		// fetch context because otherwise kubectl won't be able to talk to cluster
+		cmd := exec.Command("gcloud", "container", "clusters", "get-credentials", *gkeTestClusterName, "--project", project, locationArg, locationVal)
+		err = runCommand(fmt.Sprintf("fetching credentials from cluster %s", *gkeTestClusterName), cmd)
+		if err != nil {
+			return fmt.Errorf("failed to fetch credential from cluster: %v", err)
+		}
```

Comment:
After obtaining the credentials with get-credentials call, can you also add a check to wait for the node driver to be up and running before proceeding further. (i.e daemon set in kube-system namespace).
See this https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/deploy/kubernetes/wait-for-driver.sh#L14 and https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/master/test/k8s-integration/driver.go#L75 for example.

For the managed